cmd/skuba/cluster/init.go

```diff
@@ -32,6 +35,22 @@ type initOptions struct {
 	StrictCapDefaults bool
 }
 
+// Checks if clustername provided is correct
+func isValidClusterName(clustername string) bool {
+	// NOT a letter, a number, underscore/dash
+	reg, err := regexp.Compile("[^a-zA-Z0-9_-]+")
+	if err != nil {
+		klog.Fatalf("Error compiling regex: %s", err)
+	}
+	if reg.MatchString(clustername) {
+		return false
+	}
+	if len(clustername) == 0 {
```

Comment:
I would move this as a first step, even before compiling the regexp, is the least expensive to validate.